pallets/permission0/api/src/lib.rs

```diff
@@ -161,6 +161,29 @@ pub trait Permission0NamespacesApi<AccountId, NamespacePath> {
     fn is_delegating_namespace(delegator: &AccountId, path: &NamespacePath) -> bool;
 }
 
+pub struct WalletPermission<AccountId> {
+    pub recipient: AccountId,
+    pub r#type: WalletScopeType,
```

Comment:
This was converted by Polkadot.js to `r_type`, and in Serde, it would be just `type`. Even worse, Polkadot.js usually uses camel case, and in this case, there's this underscore `_`.

I think it's better for us to avoid using `type` or other keywords as identifiers if this is going to happen, to avoid inconsistencies in serialization/conversion at interfaces.

Alternatives: kind, variant, body.